app/component/summary/summary-row.cjsx

```diff
@@ -79,39 +80,31 @@ class SummaryRow extends React.Component
       else
         circleClass = leg.mode.toLowerCase()
 
-      legs.push (
-        <span key={i + 'a'} style={styleLine} className={leg.mode.toLowerCase()}>
-          <span key={i + 'b'} className="summary-circle #{circleClass}#{passiveClass}"></span>
-          <Icon key={i + 'c'} className={leg.mode.toLowerCase()} img={'icon-icon_' + leg.mode.toLowerCase()} />
-          {text}
-        </span>
-      )
+      legs.push <span key={i + 'a'}
+        style={styleLine}
+        className={leg.mode.toLowerCase()}>
+        <span key={i + 'b'} className="summary-circle #{circleClass}#{passiveClass}"></span>
+        <RouteNumber mode={leg.mode.toLowerCase()} text={text}/>
+      </span>
+
+      startTimeComponent = <DepartureTime
+        departureTime={leg.startTime / 1000}
+        realtime={leg.realTime}
+        currentTime={currentTime}
+        style={styleTime} />
 
       if isFirstLeg
-        legTimes.push (
-          <span key={i + 'a'} style={styleTime}>
-            {legStart.format("HH:mm")}
-          </span>
-        )
+        legTimes.push startTimeComponent
```

Comment:
Could all of these be put inside something like unless isLastLeg and not isEnoughRoomForLastLegStartTime and separate check for is isLastLeg to put the arrival time for the journey.